### PR TITLE
Substitute unwrap for ? operator

### PR DESCRIPTION
**Pull request by amadeusine**
**I'm submitting a** refactor (and fixes for test cases).

# Description

This PR addresses C-QUESTION-MARK in the roadmap for #191. The examples have been updated to accommodate for the `?` operator instead of `.unwrap` or `try!`.

# Motivation

UUID can move forward with other portions of documentation given this has been completed.

# Tests

All existing tests remain successful after changes.

# Related Issue(s)

#191

### CODE REVIEW

**Review by Dylan-DPC — DISMISSED** [review]

**Comment by Dylan-DPC** [comment]
bors: r+

**Comment by Dylan-DPC** [comment]
@kinggoesgaming can you check the travis configuration if you have time? thanks

**Comment by kinggoesgaming** [comment]
bors try

**Comment by kinggoesgaming** [comment]
An update: I have been trying to get the one heck of a mess that is our build system fixed. This might take sometime to get fixed. I want a good fix not just a system held together with duct tape.

**Review by kinggoesgaming — DISMISSED** [review]
Thank you for the PR and sorry for the long wait

File: src/lib.rs [hunk 1]
```diff
@@ -80,23 +80,25 @@
...
 //!
-//! ```ignore,rust
+//! ```rust
 //! // Note that this requires the `v4` feature enabled in the uuid crate.
 //!
 //! use uuid::Uuid;
 //!
-//! fn main() -> Result<(), Box<std::error::Error> {
+//! fn main() -> Result<(), Box<dyn std::error::Error>> {
 //!     let my_uuid = Uuid::new_v4()?;
```
Comment (kinggoesgaming) [inline src/lib.rs:99]:
```suggestion
//!     #[cfg(feature = v4")] {
//!         let my_uuid = Uuid::new_v4()?;
//!         println!("{}", my_uuid);
//!    }
```

File: src/lib.rs [hunk 2]
```diff
@@ -80,23 +80,25 @@
...
 //!     println!("{}", my_uuid);
```
Comment (kinggoesgaming) [inline src/lib.rs:100]:
```suggestion
```

**Review by kinggoesgaming — CHANGES_REQUESTED** [review]
Hey

If you can get the changes made, I will get the PR merged

**Comment by amadeusine** [comment]
Oops, prematurely pushed here. I'll fix it on my end.

**Comment by kinggoesgaming** [comment]
No probs just ping when you are ready

**Comment by kinggoesgaming** [comment]
@amadeusine any update?

**Comment by amadeusine** [comment]
Update on the changes in question: It looks like the v4 example will not pass tests under `--all-features` as 
`let my_uuid = Uuid::new_v4()?;` propagates `getrandom::Error` (which doesn't impl `std::error::Error` unless enabled). While it is possible to explicitly specify `fn main() -> Result<(), getrandom::Error>`, doing so would then fail on tests with `--no-default-features` due to getrandom being optional.

If either is acceptable, I can go ahead and push. Suggestions welcome.

**Comment by kinggoesgaming** [comment]
I think the best course of action would be to change to `ignore,rust` to ignore the test for now. The error type is expected to have changes and I think we can cover it under it.

**Comment by amadeusine** [comment]
It looks like the last change to the ignored example was already merged. Would it be fine to close the PR, then?

**Review by kinggoesgaming — APPROVED** [review]

**Pull request merged**